lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -148,23 +148,30 @@ const AbstractDialog = forwardRef<HTMLDialogElement, Props>((props, ref) => {
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       const dialogEl = dialogRef.current!;
 
-      // When tab key is pressed
-      // cycles focus on first or last focusable item (if exists)
-      // Do the same if focus is not IN the dialog to ensure focus is trapped within.
+      // Don't manage tab cycling when the focus is not within the dialog
+      // so that there is no keynav conflict when another modal is open.
+      if (!dialogEl.contains(document.activeElement)) {
+        return;
+      }
+
+      // When the tab key is pressed,
+      // cycle focus on the first or last focusable item (if exists).
+      // Do the same if focus is NOT in the dialog to ensure that the focus is trapped within.
       if (event.key === 'Tab') {
         const firstFocusableElement = getFirstVisibleFocusableElement(dialogEl);
         const lastFocusableElement = getLastVisibleFocusableElement(dialogEl);
-        const activeFocusIsNullOrDialog
-          = [document.body, dialogEl, null].includes(document.activeElement as HTMLElement);
-        const activeFocusIsInDialog = !activeFocusIsNullOrDialog && dialogEl.contains(document.activeElement);
+        const focusIsDialogElement = document.activeElement === dialogEl;
 
-        if (!activeFocusIsInDialog) {
+        // Prevent native tab cycling
+        // when focus is the dialog element, so it does not leak
+        // (In the case where there is no focusable elements inside the dialog)
+        if (focusIsDialogElement) {
```

Comment:
To me it would be slightly cleaner to think about if instead of this special case, we adjusted `getFirstVisibleFocusableElement` and `getLastVisibleFocusableElement` to return the `dialogEl` in the case that there are no other focusable elements.  That matches the behavioral model more closely: if nothing else in the modal is focusable, then the modal itself is focusable.  And then I think the correct behavior would occur automatically from the conditionals after this one (which themselves could be further cleaned up a little) and this conditional could be removed.

Reply:
> To me it would be slightly cleaner to think about if instead of this special case, we adjusted `getFirstVisibleFocusableElement` and `getLastVisibleFocusableElement` to return the `dialogEl` in the case that there are no other focusable elements. That matches the behavioral model more closely: if nothing else in the modal is focusable, then the modal itself is focusable. And then I think the correct behavior would occur automatically from the conditionals after this one (which themselves could be further cleaned up a little) and this conditional could be removed.

Thanks! That simplified things a lot and definitely cleaner: c9979e1e4fe14938a2b279e719635ef644f15bdf

Reply:
Also removed early exit here: 73be804f6399907868d8c567432252bad4f88b71